### DISCUSSION TITLE
AzuraCast Licensing and Third-Party Hosts/"Resellers"

### DISCUSSION THREAD

**Discussion by BusterNeece** [AzuraCast/AzuraCast#4071] [General]
Hello everyone! I'm the lead developer and maintainer of AzuraCast.

As AzuraCast has grown in popularity, a number of third-party hosting/"reseller" services have popped up offering managed AzuraCast hosting via their own servers. The purpose of this thread is to clarify what our position is about these services and answer any big questions folks might have.

If you don't feel like reading the whole thing, here's a summary:

 - **AzuraCast itself is both free (as in libre) and free-of-charge (as in gratis) software.** We do not sell it via any channels. You can [install it yourself](https://www.azuracast.com/install/) at no cost any time you like.

 - **We do not officially support or sponsor any third-party hosts.** Hosts aren't required to donate to our software, and many don't, so there is no way of knowing if you are supporting our project directly without [supporting our project directly](https://www.azuracast.com/donate.html).

 - **There is nothing we as a team can do about third-party hosts.** Our open-source license is permissive by design, but no open-source license we could select would prevent commercial resale of our software, as then it's no longer "open-source". I don't personally like the dynamic this creates, but there's nothing much we can do about it.

 - **If you purchase AzuraCast hosting from a third-party company, contact them for support.** Many third-party resellers are directing their customers to this GitHub for support. Please contact the company directly to resolve any issues, as we have no relationship with the hosting company. We're just making the radio software and, again, giving it away.

---

### AzuraCast and Open Source

As we've mentioned before, AzuraCast is free and open-source on purpose, and we plan to remain this way for the foreseeable future. Despite its shortcomings, we are big believers in the concept of open-source software. We're also aware that for many of our users, ours is a hackable, customizable, flexible alternative to the monolithic, closed-source software that came before us, and we think that's pretty cool and we want to stay that way.

The permissiveness of this license has led to some headaches for our team. Basically, while we don't sell the software ourselves, that hasn't stopped people from using our hard work and selling it themselves, and there's absolutely nothing we can do about that without becoming software that's no longer considered "free and open-source".

Recently, we evaluated the idea of relicensing AzuraCast under a "dual-license" scheme where we would make the software available as free and open-source via the Affero GPL (AGPL) for most users and a commercial license for users who didn't want to share their modifications to the software. While we saw a bunch of support from our users for doing this, there also were some obstacles that made it impractical to implement:
 - Selling commercial licenses would require that we turn into a legal commercial entity (either for-profit or non-profit), which involves a lot of paperwork and bureaucracy that we don't have the extra personnel, time or energy to do;
 - While the AGPL would require the resharing of core changes to our software, most hosts/resellers don't actually modify the software at its core, instead using either plugins or custom branding, neither of which would be covered by the AGPL; and
 - A majority of hosting companies exist outside the United States and it would thus be basically impossible for us to enforce any licensing restrictions on them, especially given our small size and lack of a legal team.

With those limitations in mind, we're currently sticking with our existing Apache 2.0 license. Our contributors have been very supportive throughout this process, and we've established a Contributor License Agreement (CLA) in the event we do decide to relicense in the future, but for now we're sticking with the status quo.

### Third-Party Hosts/Resellers

Since AzuraCast has grown in popularity, a _bunch_ of companies have sprung up offering managed AzuraCast hosting services, where they'll host your radio station, give you all the necessary permissions for a single (or multiple) stations, and then let you do the rest.

To be entirely fair to these companies, some of them are excellent (in my opinion). They'll provide you direct support through their own infrastructure, relay any technical issues to us with all the proper logs and documentation, and they also support our project through significant recurring donations.

It's the ones that aren't excellent that cause me the most headache. Some companies don't provide any support to their users or support AzuraCast itself, yet when issues arise, they direct their users to our GitHub pages for support. Worse yet, they don't provide any support links _at all_, and angry users end up coming to our GitHub and ranting about how they've lost money. We didn't make that money, and we have no relationship with that company; we just build and maintain the (again, free) software they're using.

This latter group has created a significant burden on our support structure without supporting us for the extra time we're spending doing that support. This would be exploitative and wrong even on a regular day, but when we're under the intense emotional stress of a pandemic and just trying to keep the hobby we love going strong, it takes an even heavier toll than usual.

### What We're Doing

When these issues come up, a lot of our users are hoping we have a definitive recourse for it, that there's some sort of firm, decisive action we can take that will stop the bad people from commercializing our work without compensating us for it.

Well, the truth is...we _don't_ have a definitive recourse for it. And that's what makes me angriest of all.

We're stuck in the same catch-22 that basically every free and open-source project is stuck in:

 - If we stay free and open-source, there will never be anything we can do to prevent commercial use or resale from any company of any size. Every open-source license out there allows commercial use, and every license that doesn't isn't an OSI-certified "open-source" license. Thus, open-source is, by design, open for both friendly use _and_ exploitation, _yet_

 - If we modify our license that in any way deviates from OSI-approved open-source licenses, we'll immediately lose access to a ton of things that we depend on to operate. Our DigitalOcean hosting, our JetBrains software, our CrowdIn translations, our Netlify homepage would all no longer be provided to us for free, and we absolutely don't have the capital to support paying for all of those things ourselves.

This carrot-and-stick dilemma means that the only way we can survive as an open-source project is to make ourselves completely open to the same exploitation that drains our resources, energy and motivation. If you've seen my Twitter, you know this is not something I'm happy about, but it's the status quo of open-source as it is today.

All we can do is enforce our own terms and standards in the places we control, and this GitHub is one of them. If your issue is clearly one that should be addressed via a third-party host/reseller instead of us, we'll close the issue and let you know this.

### What About DigitalOcean/Linode?

As a footnote, some of you might notice that we have one-click installers on our [installation page](https://www.azuracast.com/install/) for two hosting companies (DigitalOcean and Linode) and wonder if they're included in this discussion about third-party hosts/resellers.

The answer is **no** and the reason is simple: all the one-click installers are doing is just configuring a standard VPS to have AzuraCast on it out of the box. These hosting companies are just providing generic hosting services that you could go sign up for yourself, and the sole purpose of our installers is just to make the initial spin-up process a little easier. After that, it's up to you to maintain your own AzuraCast instance, just as if you had hosted it yourself.

I realize most users are aware of this distinction already, but I wanted to go ahead and answer that question preemptively.

**Comment by andrewfer000** [comment]
Hey Eagle, I kinda had a feeling you wouldn't want to go through the dual licensing since in reality it would be more of a pain then what it would be worth to you and the project in general. I was hoping still that you would at least try and use some kind of OSI approved semi-copyleft license such as the MPL just to use a better license but it is what it is. I wanted to let you know about some good ways to continue possibly generate revenue and not burn yourself out while also not having to force a more restrictive license on the project. I have some tips for you if you want some advice.

1. Check out some of the methods the Pterodactyl Game Panel team are using to generate income and get more devs on borard while still using the very weak MIT license. Things such as Discord perks, third-party partnerships, and maybe even the occasional Hackathon live stream that can get more donators and help grow the official AzuraCast community in hopes to get more sponsors and developers. Be more active in the greater tech community such as Reddit so more people will see your software. If you talk about AzuraCast at the right places you will get a lot of positive attention.        

2. Use the MPL, GPL, or AGPL, but instead of doing official dual licensing you would just "look the other way" for contributors and active donors. I can't remember the name of the software that did something like this but I remember it from somewhere. Just DO NOT use a Non-OSI approved license (like you already mentioned).  

3. Create a "feature bounty" program where donors can pick features to be implemented by donating to that specific item or mentioning it in their GitHub sponsor. (In my opinion this and #1 are the best together).

4. Give yourself a break. AzuraCast is a huge project but at the same time it is very stable. (at least when I used it) and doesn't have super major issues that need to be fixed right this minute. Focus on yourself and make sure that you enjoy working on the project. That's what Open Source is all about :) 

5. Give incentives to third-party resellers who would pay lots of money and/or contribute back code to the project. Incentives such as more control over new feature decisions and direct support from you can be a major hit with the third parties.

6. Make a "Wall of Shame" for bad companies breaking the rules and/or not being friendly in hopes they would see it and then "right their wrongs".   

7. Just remember that weather your code is public domain, Apache, or AGPL people will still take advantage and break the rules. While that sucks and is very morally wrong its just a fact of Open Source and we can only try to make it better ourselves.   

Have a good day my friend, Lmk if you have any questions about any of this.

- Andrew

**Comment by xaoseric** [comment]
I would suggest the following.

1. Keep the open source version for personal use.
2. Sell an edition for hosting providers for like $10/mo per server or something with features like users being able to create packages which would allow for resellers in the panel, and priority support via tickets.

**Comment by ImaCrea** [comment]
Maybe check what's doing Write.as -> https://write.as/writefreely
It's surely a lot of work but one of the best inspiration I found out there trying to fix that. Maybe try to speak with them?

**Comment by gusaus** [comment]
This article about [CiviCRM](https://civicrm.org/blog/michaelmcandrew/paths-to-sustainability) may provide some inspiration and guidance with regards to sustaining the AzuraCast project and a core team.

One of projects referenced provides some interesting hosting models -
- Free - https://github.com/tryghost/ghost
- Fully managed - https://ghost.org/pricing/
- Self hosted /w support - https://valet.ghost.org/

These CiviCRM sustainability initiatives, in addition to paid bounties/feature requests/bug fixes mentioned in this issue, may also work for AzuraCast - 

[Make It Happen (MIH)](https://civicrm.org/make-it-happen) - campaigns to get or support new features that you’d like to see into the project but which hasn’t been included within core development
[Partner Program](https://civicrm.org/become-a-partner) - for service providers that want to financially support the project
[Member Program](https://civicrm.org/become-a-member) - for end users that want to do the same.

Funds from most, if not all, of these models could be collected/distributed via the [AzuraCast collective](https://opencollective.com/azuracast) and fiscal host/sponsor [Open Source Collective](https://opencollective.com/opensource) already set up.

Apologies for adding a closed issue - possibly we can set this up as a discussion? https://github.com/AzuraCast/AzuraCast/discussions

**Reply by gusaus** [reply]
I set up this last bit as more of a catch-all discussion about sustaining AzuraCast https://github.com/AzuraCast/AzuraCast/discussions/3994

Good to see this post as a discussion rather than a closed issue.

**Comment by batsonjay** [comment]
Dries Buytaert, [Drupal](https://drupal.org) project lead since 2001, wrote [a thoughtful and thorough discussion](https://dri.es/balancing-makers-and-takers-to-scale-and-sustain-open-source) about this tension often seen in open source. (It has existed in the Drupal community for a long time.) His post proposes several ways to handle the "Makers vs Takers" challenge you are essentially describing here (and which is also related to the sustaining AzuraCast issue #3994).

I encourage you (Eagle) to read it. I'm happy to discuss this more, too, as I see AzuraCast being important, and going thru many of the same growing pains Drupal did. (Not sure how to best handle such a discussion.)
